zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/FlowNodeInstanceQueryControllerTest.java

```diff
@@ -98,4 +163,175 @@ void shouldSearchFlownodeInstancesDecisionWithEmptyBody() {
+             { "field": "processInstanceKey", "order": "ASC" },
+             { "field": "processDefinitionKey", "order": "ASC" },
+             { "field": "bpmnProcessId", "order": "ASC" },
+             { "field": "startDate", "order": "DESC" },
+             { "field": "endDate", "order": "DESC" },
+             { "field": "flowNodeId", "order": "ASC" },
+             { "field": "type", "order": "ASC" },
+             { "field": "state", "order": "ASC" },
+             { "field": "incidentKey", "order": "ASC" },
+             { "field": "tenantId", "order": "ASC" }
+           ]
+         }
+        """;
+    webClient
+        .post()
+        .uri(FLOW_NODE_INSTANCES_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(flowNodeInstanceServices)
+        .search(
+            new FlowNodeInstanceQuery.Builder()
+                .sort(
+                    new FlowNodeInstanceSort.Builder()
+                        .flowNodeInstanceKey()
+                        .asc()
+                        .processInstanceKey()
+                        .asc()
+                        .processDefinitionKey()
+                        .asc()
+                        .bpmnProcessId()
+                        .asc()
+                        .startDate()
+                        .desc()
+                        .endDate()
+                        .desc()
+                        .flowNodeId()
+                        .asc()
+                        .type()
+                        .asc()
+                        .state()
+                        .asc()
+                        .incidentKey()
+                        .asc()
+                        .tenantId()
+                        .asc()
+                        .build())
+                .build());
+  }
+
+  @Test
+  void shouldGetFlowNodeInstanceByKey() {
+    when(flowNodeInstanceServices.getByKey(any(Long.class))).thenReturn(GET_QUERY_RESULT);
+    // when / then
+    webClient
+        .get()
+        .uri(FLOW_NODE_INSTANCES_URL + "23")
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
+        .expectBody()
+        .json(EXPECTED_GET_RESPONSE);
+
+    verify(flowNodeInstanceServices).getByKey(23L);
+  }
+
+  @Test
+  void shouldThrowNotFoundIfKeyNotExistsForGetFlowNodeInstanceByKey() {
+    when(flowNodeInstanceServices.getByKey(any(Long.class))).thenThrow(new NotFoundException(""));
+    // when / then
+    webClient
+        .get()
+        .uri(FLOW_NODE_INSTANCES_URL + "5")
+        .exchange()
+        .expectStatus()
+        .is5xxServerError()
```

Comment:
🔧  For this case I see you are throwing a `new NotFoundException("")` but testing if the error status is 500

With the update, the Rest Error is treating properly NotFoundException, this test should be revisited.